src/Config/RuntimeConfigLoader.cs

```diff
@@ -112,9 +124,11 @@ public static JsonSerializerOptions GetSerializationOptions()
         options.Converters.Add(new EnumMemberJsonEnumConverterFactory());
         options.Converters.Add(new RestRuntimeOptionsConverterFactory());
         options.Converters.Add(new GraphQLRuntimeOptionsConverterFactory());
-        options.Converters.Add(new EntitySourceConverterFactory());
+        options.Converters.Add(new EntitySourceConverterFactory(replaceEnvVar));
+        options.Converters.Add(new EntityGraphQLOptionsConverterFactory(replaceEnvVar));
+        options.Converters.Add(new EntityRestOptionsConverterFactory(replaceEnvVar));
         options.Converters.Add(new EntityActionConverterFactory());
-        options.Converters.Add(new StringJsonConverterFactory());
+        options.Converters.Add(new StringJsonConverterFactory(replaceEnvVar));
```

Comment:
Rather than having to pass down this variable, would it be easier to not include this converter if the environment variable is set?

I was thinking that you'd just put a check around this line of _if flag disables replacement, don't include_ rather than having to have logic within the converters.

Reply:
You mean in those cases rely on default String converter? Are we losing any other functionality because of which we needed this custom converter? Or was the custom converter only needed for environment variable replacement?

Reply:
Yes, the only job of that `JsonConverter` is to apply the env-var replacement rules.

Reply:
makes sense. fixed.